snapshot/git/gitdb/bundlestore.go

```diff
@@ -0,0 +1,200 @@
+package gitdb
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+
+	snap "github.com/scootdev/scoot/snapshot"
+	"github.com/scootdev/scoot/snapshot/bundlestore"
+)
+
+// BundlestoreConfig defines how to talk to Bundlestore
+type BundlestoreConfig struct {
+	Store bundlestore.Store
+}
+
+type bundlestoreBackend struct {
+	cfg *BundlestoreConfig
+}
+
+const bundlestoreIDText = "bs"
+
+// "bs-gc-<bundle>-<stream>-<sha>"
+
+func (b *bundlestoreBackend) parseID(id snap.ID, kind snapshotKind, extraParts []string) (snapshot, error) {
+	if len(extraParts) != 3 {
+		return nil, fmt.Errorf("cannot parse snapshot ID: expected 5 parts in bundlestore id: %s", id)
+	}
+	bundleName, streamName, sha := extraParts[0], extraParts[1], extraParts[2]
+
+	if err := validSha(sha); err != nil {
+		return nil, err
+	}
+
+	return &bundlestoreSnapshot{kind: kind, sha: sha, bundleName: bundleName, streamName: streamName}, nil
+}
+
+func (b *bundlestoreBackend) upload(s snapshot, db *DB) (snapshot, error) {
+	// We only have to upload a localSnapshot
+	switch s := s.(type) {
+	case *tagsSnapshot:
+		return s, nil
+	case *streamSnapshot:
```

Comment:
Tags vs. Streams:

We will have the stream in every GitDB. But each tag will only be in some GitDBs. So when we create a bundle, it will base off a stream, which it assumes is easy to get, but not a tag.

Stream uploading:

something outside scoot pushes to master. On the macbook, they'll fetch outside scoot (just call git fetch), and then when they upload, GitDB will realize if the commit is on a stream and if so use that. (that's to be implemented)